components/movementsensor/replay/replay.go

```diff
@@ -454,7 +513,79 @@ func addGRPCMetadata(ctx context.Context, timeRequested, timeReceived *timestamp
 	return nil
 }
 
-// extractDataAndMetadata retrieves the next cached data and removes it from the cache. It assumes the write lock is being held.
+func (replay *replayMovementSensor) setProperty(method method, supported bool) error {
+	switch method {
+	case position:
+		replay.properties.PositionSupported = supported
+	case linearVelocity:
+		replay.properties.LinearVelocitySupported = supported
+	case angularVelocity:
+		replay.properties.AngularVelocitySupported = supported
+	case linearAcceleration:
+		replay.properties.LinearAccelerationSupported = supported
+	case compassHeading:
+		replay.properties.CompassHeadingSupported = supported
+	case orientation:
+		replay.properties.OrientationSupported = supported
+	default:
+		return errors.New("can't set property, invalid method: " + string(method))
+	}
+	return nil
+}
+
+// attemptToInitializeProperty will try to update the cache for the provided method.
+func (replay *replayMovementSensor) attemptToInitializeProperty(ctx context.Context, method method) (bool, error) {
+	initializedProperty := false
+	if replay.closed {
+		return initializedProperty, errSessionClosed
+	}
+	if err := replay.updateCache(ctx, method); err != nil && !strings.Contains(err.Error(), ErrEndOfDataset.Error()) {
+		return initializedProperty, errors.Wrap(err, "could not update the cache")
+	}
+	if len(replay.cache[method]) != 0 {
+		if err := replay.setProperty(method, true); err != nil {
+			return initializedProperty, errors.Wrap(err, "could not set property")
+		}
+		initializedProperty = true
+	}
+	return initializedProperty, nil
+}
+
+// initializeProperties will set the properties by repeatedly attempting to poll data from all the methods
+// until at least one of them returns data. The properties are set to `true` for the endpoints that
+// returned data.
+func (replay *replayMovementSensor) initializeProperties(ctx context.Context) error {
+	var initializedAtLeastOneProperty, initializedProperty bool
+	var err error
+	// Repeatedly attempt to poll data from the movement sensor for each method until at least
+	// one of the methods receives data.
+	for {
+		if !goutils.SelectContextOrWait(ctx, initializePropertiesLoopWaitTime) {
+			return ctx.Err()
+		}
+		for _, method := range methodList {
```

Comment:
[note] If you still want to make a call to each method after the first has been confirmed then you can, although with the 1sec loop wait it may be unluckily that data was uploaded mid properties check.